packages/hardhat/scripts/generateTsAbis.ts

```diff
@@ -14,7 +14,7 @@ function main() {
     fs.mkdirSync(TARGET_DIR);
   }
   fs.writeFileSync(
-    `${TARGET_DIR}hardhat_contracts.ts`,
+    `${TARGET_DIR}deployed_contracts.ts`,
     prettier.format(`export default {${fileContent}} as const;`, { parser: "typescript" }),
```

Comment:
`NITPICK:`

Maybe we do : 
```ts
 fs.writeFileSync(
    `${TARGET_DIR}deployed_contracts.ts`,
    prettier.format(`const contracts = {${fileContent}} as const; \n\n export default contracts`, {
      parser: "typescript",
    }),
  );
```

#### Reason : 

The initial state of `deployed_contracts.ts` is :
```ts
const contracts = null;
export default contracts;
```

Final state of `deplyoyed_contracts.ts` after running `yarn deploy` : 
```ts
export default {
  //...
} as const
```

It would be great if keep final state similar to inital, just to keep it consistent 

Final  state : 
```ts
const contracts = {
// ...
} as const 

export default contracts;
```

We also get the added benefit of auto import which might be helpful if people want to get abi : 
![Screenshot 2023-04-04 at 4 13 02 PM](https://user-images.githubusercontent.com/80153681/229769383-16c96234-31cc-4d2d-822f-0643dfe43b1d.jpg)


Maybe we can change name from `contracts` -> `contractsAbis` ?

Reply:
The auto-import would be very helpful indeed :+1: 

No strong opinion on the name, but kinda make sense, not sure what users would find the most intuitive.

Reply:
> It would be great if keep final state similar to inital, just to keep it consistent

I buy that! Fixed on 819b4df1e4dca9ac0535b94164a536328633f240

> We also get the added benefit of auto import which might be helpful if people want to get abi :
> The auto-import would be very helpful indeed

I think we should recommend people to use the `useDeployedContractInfo` hook.

> Maybe we can change name from contracts -> contractsAbis ?

I could see `deployedContracts`. But `contractsAbis` doesn't seem right, since the file stores more than that (deployed network, deployed address)

Reply:
> I think we should recommend people to use the useDeployedContractInfo hook.

yes yes 💯

> I could see deployedContracts. But contractsAbis doesn't seem right, since the file stores more than that (deployed network, deployed address)

Agreed, I was just suggesting a different name because it has conflicts with `~/utils/scaffold-eth/contracts`,  BTW I love `deployedContracts` name

Reply:
Changed the var name and also the file name.

`deployedContracts.ts` is more aligned to our naming conventions, right?

Reply:
Lol don't know why `deployedContracts.ts` makes me feel uneasy 😂 maybe I am habitat to seeing a file with `_` in the generated directory. 

But yeah `deployedContracts.ts` seems more aligned 🙌